firestore/src/main/java/com/firebase/ui/firestore/ObservableSnapshotArray.java

```diff
@@ -18,4 +19,11 @@
     public ObservableSnapshotArray(@NonNull SnapshotParser<T> parser) {
         super(new CachingSnapshotParser<>(parser));
     }
+
+    /**
+     * Method for changing/updating the {@link Query} in existing adapter.
+     *
+     * @param newQuery is a new updated query.
+     */
+    public abstract void updateQuery(@NonNull Query newQuery);
```

Comment:
Same question here as on the `database` version.

Reply:
The reason is also the same as I explained above.